lib/config/enabled-managers.ts

```diff
@@ -0,0 +1,22 @@
+import is from '@sindresorhus/is';
+import { logger } from '../logger';
+import { getManagerList } from '../manager';
+import type { ManagerConfig, RenovateConfig } from './types';
+
+export function applyEnabledManagersFilter(
+  config: RenovateConfig
+): RenovateConfig {
+  const { enabledManagers = [] } = config;
+  if (is.nonEmptyArray(enabledManagers)) {
+    logger.debug({ enabledManagers }, 'Applying enabled managers filtering');
+    const enabledSet = new Set([...enabledManagers]);
+    for (const managerName of getManagerList()) {
+      const manager = config[managerName] as ManagerConfig;
+      if (manager) {
+        manager.enabled = enabledSet.has(managerName);
+      }
+    }
+    enabledManagers.splice(0, enabledSet.size);
```

Comment:
What's the purpose of this line?

Reply:
Reset the array, actually it's better to not do this